### PR TITLE
Milan/renew temprary url

### PR DESCRIPTION
**Pull request by milancausecode**

### CODE REVIEW

**Comment by HardikModha** [comment]
[(3) FU - Rewrite the renewTemporaryUrl method to include all FileUrls.](https://trello.com/c/V8P2WyPO/445-3-fu-rewrite-the-renewtemporaryurl-method-to-include-all-fileurls)

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @since 3.
+ */
+@Transactional
+class UFileTemporaryUrlRenewerService {
+
+    UtilitiesService utilitiesService
+
+    void renewTemporaryURL(boolean forceAll = false) {
+        CDNProvider[] allProviders = CDNProvider.values()
+        CDNProvider[] providersToExclude = [CDNProvider.RACKSPACE, CDNProvider.LOCAL]
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Can be simplified with 

```
CDNProvider.findAll {
    !(it in [CDNProvider.RACKSPACE, CDNProvider.LOCAL]) 
}
```

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        (allProviders - providersToExclude).each { CDNProvider cdnProvider ->
+            CDNFileUploader cDNFileUploader = utilitiesService.getProviderInstance(cdnProvider.name())
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Rename to `cdnFileUploaderInstance` and the method throws `ProviderNotFoundException`.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+            UFileTemporaryUrlRenewer temporaryUrlRenewer =
+                    new DefaultUFileTemporaryUrlRenewer(cdnProvider, cDNFileUploader, forceAll)
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: `IllegalArgumentException`

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * This method is used for dynamically instantiating the CDNFileUploader class based on the Provider.
+     *
+     * @param providerName The name of the provider.
+     * @return Instance of the CDNFileUploader class.
+     *
+     * @author Nikhil Sharma
+     * @since 2.4.9
+     */
+    CDNFileUploader getProviderInstance(String providerName) {
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: throws `ProviderNotFoundException`

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+        try {
+            return grailsApplication.classLoader.loadClass(providerClassName)?.newInstance()
+        } catch (ClassNotFoundException e) {
+            log.debug 'Could not find Provider class', e
+            throw new ProviderNotFoundException("Provider $providerName not found.", e)
+        }
+    }
+
+    @SuppressWarnings('JavaIoPackageAccess')
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: Javadocs

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    String getNewTemporaryDirectoryPath() {
+        String tempDirectoryPath = FileUploaderUtils.baseTemporaryDirectoryPath + UUID.randomUUID().toString() + '/'
+        File tempDirectory = new File(tempDirectoryPath)
+        tempDirectory.mkdirs()
```
Comment (ankit-agrawal11) [inline grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: Line number 52 and 53 are not needed. Please visit `FileUploaderUtils`

File: src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            int maxResultsInOneIteration = 100) {
+
+        checkArgument(cdnProvider != null, 'CDNProvider can not be null')
+        checkArgument(cdnFileUploader != null, 'CDNFileUploader can not be null')
+        checkArgument(maxResultsInOneIteration > 0, 'maxResultsInOneIteration can\'t be negative or zero')
+
+        this.cdnProvider = cdnProvider
+        this.cdnFileUploader = cdnFileUploader
+        this.forceAll = forceAll
+        this.maxResultsInOneIteration = Math.min(maxResultsInOneIteration, 1000)
```
Comment (ankit-agrawal11) [inline src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy]: 1000 is too big a number.

File: src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    @SuppressWarnings('CannotModifyReference') // Method just saves the given instance after updating the properties
+    private void updateExpirationPeriodAndUrl(UFile uFile) {
+        long expirationPeriod = getExpirationPeriod(uFile.fileGroup)
+
+        uFile.path = cdnFileUploader.getTemporaryURL(uFile.container, uFile.fullName, expirationPeriod)
+        uFile.expiresOn = new Date(new Date().time + expirationPeriod * 1000)
+
+        if (NucleusUtils.save(uFile, true)) {
+            log.debug "New URL for $uFile [$uFile.path] [$uFile.expiresOn]"
+        }
```
Comment (ankit-agrawal11) [inline src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy]: May be you can throw an exception here if save call returns false. Catch and log it in line number 98.

File: src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import spock.lang.Specification
+
+/**
+ * Unit tests for {@link DefaultUFileTemporaryUrlRenewer}
+ *
+ * @author Milan Savaliya
+ * @since 3.1.3
+ */
+@TestFor(UFileTemporaryUrlRenewerService)
+class DefaultUFileTemporaryUrlRenewerServiceSpec extends Specification {
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy]: Service name does not include `Default`. The test class name should be `UFileTemporaryUrlRenewerServiceSpec`.

File: src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        DefaultUFileTemporaryUrlRenewer renewer = Mock(DefaultUFileTemporaryUrlRenewer)
+        renewer.renew() >> { /*do nothing*/ }
+        GroovyMock(DefaultUFileTemporaryUrlRenewer, global:true)
+        new DefaultUFileTemporaryUrlRenewer(_ , _ ,_) >> renewer
+
+        when:
+        service.renewTemporaryURL(true)
+
+        then:
+        noExceptionThrown()
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy]: Instead of mocking the `DefaultUFileTemporaryUrlRenewer` if we pass some invalid arguments while creating instances, it will throw an exception.

For success case, you can mock the log and check if it gets invoked.

File: src/test/groovy/com/causecode/fileuploader/FileUploaderServiceSpec.groovy [hunk 11]
```diff
@@ -481,6 +487,13 @@ class FileUploaderServiceSpec extends BaseFileUploaderServiceSpecSetup {
         UFile uFileInstance = UFile.build(type: UFileType.CDN_PUBLIC)
         assert uFileInstance.type == UFileType.CDN_PUBLIC
 
+        and: 'Mocked getProviderInstance method'
+        1 * service.utilitiesService.getProviderInstance(_) >> { String provider ->
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/FileUploaderServiceSpec.groovy]: If the method gets called only once, there is no point defining a return value for second call.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        e.message == 'Provider test not found.'
+
+        when: 'getProviderInstance method is called and class exist'
+        service.getProviderInstance('Amazon')
+
+        then: 'No exception is thrown'
+        noExceptionThrown()
+    }
+
+    void "test getNewTemporaryDirectoryPath method"() {
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Please complete the sentences. It's really not clear what are you trying to test here. Whether it's a success case or failure and what are the expectations.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        when:
+        service.newTemporaryDirectoryPath
+        then:
+        noExceptionThrown()
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Does the call return something? Please check for the response.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    void "test getTempFilePathForMultipartFile"() {
+        when:
+        service.getTempFilePathForMultipartFile('Foo', '.pdf')
+        then:
+        noExceptionThrown()
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Here. Also, please add statements to make the test more clear.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+        dir.mkdir()
+        dir.deleteOnExit()
+
+        when: 'instance is of simpleFile'
+        File f = new File('./test')
+        f.deleteOnExit()
+        service.moveFile(f, dir.absolutePath)
+
+        then:
+        noExceptionThrown()
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Can you check if the file exists at the new destination?

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+
+        when: 'getResultListFromOffset method is called'
+        List<UFile> resultUFiles = new DefaultUFileTemporaryUrlRenewer(
+                CDNProvider.GOOGLE,
+                fileUploader,
+                true,
+                5
+        ).getResultListFromOffset(5)
+
+        then: 'The returned resultUFiles must matches with the expected list of UFiles'
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: `must match the expected`

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+        new DefaultUFileTemporaryUrlRenewer(
+                CDNProvider.GOOGLE,
+                fileUploader,
+                true,
+                5
+        ).processResultList(uFiles)
+
+        then: 'path and expiresOn properties for the given files should be updated'
+        uFiles.every { UFile uFile ->
+            return uFile.path == 'Dummy Url' && uFile.expiresOn > new Date()
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: Can't we use assert here?

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+        }
+    }
+
+    void "test that updateExpirationPeriodAndUrl method updates the expiration date and url for the given UFile"() {
```
Comment (ankit-agrawal11) [inline src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: This one matches with the test case just above it. The only difference I see is the list and a single instance. Can we use Unroll?

File: grails-app/controllers/com/causecode/fileuploader/FileUploaderController.groovy [hunk 19]
```diff
@@ -8,6 +8,8 @@
 package com.causecode.fileuploader
 
 import grails.util.Environment
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/controllers/com/causecode/fileuploader/FileUploaderController.groovy]: Removed.

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy [hunk 20]
```diff
@@ -33,13 +35,15 @@ class DailyJob {
 
         log.info 'Started executing DailyJob..'
 
-        fileUploaderService.renewTemporaryURL()
-        fileUploaderService.moveFailedFilesToCDN()
+        UFile.withNewSession {
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: It shouldn't work with the approach you suggested.

File: grails-app/services/com/causecode/fileuploader/FileUploaderService.groovy [hunk 21]
```diff
@@ -119,13 +83,14 @@ class FileUploaderService {
                 tempFile = file
             } else {
                 if (file instanceof MultipartFile) {
-                    tempFile = getTempFilePathForMultipartFile(fileData.fileName, fileData.fileExtension)
-                    file.transferTo(tempFile)
+                    tempFile = utilitiesService.getTempFilePathForMultipartFile(fileData.fileName,
+                            fileData.fileExtension)
+                    tempFile ? file.transferTo(tempFile) : void
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/FileUploaderService.groovy]: Done.

[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/FileUploaderService.groovy]: Done.

File: grails-app/services/com/causecode/fileuploader/FileUploaderService.groovy [hunk 22]
```diff
@@ -119,13 +83,14 @@ class FileUploaderService {
...
                 }
             }
 
             // Delete the temporary file when JVM exited since the base file is not required after upload
-            tempFile.deleteOnExit()
+            tempFile?.deleteOnExit()
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/FileUploaderService.groovy]: Removed.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+import com.causecode.fileuploader.cdn.CDNFileUploader
+import com.causecode.fileuploader.util.renewer.DefaultUFileTemporaryUrlRenewer
+import com.causecode.fileuploader.util.renewer.UFileTemporaryUrlRenewer
+import grails.transaction.Transactional
+
+/**
+ * Service which communicates with UFileTemporaryUrlRenewers.
+ *
+ * @author Milan Savaliya
+ * @since 3.
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Fixed.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+ */
+@Transactional
+class UFileTemporaryUrlRenewerService {
```
[the remark this answers is not in the document]
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Changed to `TemporaryUrlRenewerService`. Apart from that, I don't see any issues having a service class with a single method as long as it is not violating any design principle.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy (hunk 1 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: No change.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy (hunk 2 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Done.

File: grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy (hunk 3 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UFileTemporaryUrlRenewerService.groovy]: Done.

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy (hunk 4 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: Done.

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy (hunk 5 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: Added.

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy (hunk 6 above)
Comment (HardikModha) [reply grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: It is needed. The code in `FileUploaderUtils` only creates a parent directory. Here we are also creating the temporary child directory.

File: src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy (hunk 7 above)
Comment (HardikModha) [reply src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy]: UFile instances are not that big and 1000 is not that big number it will hardly occupy 1-2MB when loaded into the memory.

File: grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy (hunk 6 above)
Comment (ankit-agrawal11) [reply grails-app/services/com/causecode/fileuploader/UtilitiesService.groovy]: Sorry, my bad. 👍

File: src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy (hunk 8 above)
Comment (HardikModha) [reply src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy]: Changed method to return boolean.

File: src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy (hunk 9 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy]: Fixed.

File: src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy (hunk 10 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/DefaultUFileTemporaryUrlRenewerServiceSpec.groovy]: Added a failure case.

File: src/test/groovy/com/causecode/fileuploader/FileUploaderServiceSpec.groovy (hunk 11 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/FileUploaderServiceSpec.groovy]: Done.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy (hunk 12 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Done.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy (hunk 13 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Done.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy (hunk 14 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Added.

File: src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy (hunk 15 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy]: Done.

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy (hunk 16 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: Fixed.

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy (hunk 17 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: Why do we need to use assert? This works.

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy (hunk 18 above)
Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: It's better not to combine test cases for multiple methods inside a single test case.

Comment (ankit-agrawal11) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: Right. My bad. Please ignore the suggestion.

File: src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy (hunk 17 above)
Comment (ankit-agrawal11) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: I am not sure if the condition is being validated. Maybe you can try to change it to `uFile.path != 'Dummy Url'` and see if the test fails. Just to be sure.

Comment (HardikModha) [reply src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy]: The condition is being evaluated as we are using `every`. If we had used `each` then your doubt would have become real. 😃

File: changelog.md [hunk 25]
```diff
@@ -1,5 +1,10 @@
 # ChangeLog
 
+## Version 3.1.3 [Unreleased]
+
+### Changed
+- RenewTemprary URL Refactored to new service and now handles all the files
```
Comment (shivamcausecode) [inline changelog.md]: RenewTemprary URL method refactored to new service and now handles all the files at once or something better.

File: code-climate-codenarc/CanNotModifyReferenceRule.groovy [hunk 26]
```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2017-Present, Niteo Consulting Private Limited, India.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are not permitted.
+ */
```
Comment (shivamcausecode) [inline code-climate-codenarc/CanNotModifyReferenceRule.groovy]: Why is this here? This should not be present in the open source plugin. Remove it's all occurrences.

File: src/test/groovy/com/causecode/fileuploader/util/FileUploaderUtilsSpec.groovy [hunk 27]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, CauseCode Technologies Pvt Ltd, India.
```
Comment (shivamcausecode) [inline src/test/groovy/com/causecode/fileuploader/util/FileUploaderUtilsSpec.groovy]: 2011-Present, Check in all newly added files.

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy (hunk 20 above)
[the remark this answers is not in the document]
Comment (ankit-agrawal11) [reply grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: Okay. That seemed to be the only difference between two services and since it was working earlier, I thought it might work that way.
